### DISCUSSION TITLE
Abstract Relationships: Using ForeignKey(-esque) Patterns on Abstract Models

### DISCUSSION THREAD

**Discussion by gravy-jones-locker** [sqlalchemy/sqlalchemy#10750] [Usage Questions] [answered]
Hi,

I love using SQLAlchemy and am incredibly grateful for all the work that has gone into it.

However, there is one feature which I cannot find documentation for anywhere, but which seems intuitive. I would like to be able to define ORM `relationship` fields against abstract models. Importantly, these are not models which I intend to use as parents in any kind of inheritance structure. Instead, they are objects which make sense in my application but which it does not make sense to persist.

Naturally I could define whatever `primaryjoin` or similar arguments are necessary on the relationship itself. But as it stands I cannot find any way to make this work. Instead the representational string for the relationship prints the instance details and 'no key' - e.g.:

`<Relationship at 0x1120aecb0; no key>`

If I try putting a placeholder `__tablename__` on the model (rather than using `__abstract__ = True`) then the relationship loads as `None`

If there are obvious/important structural reasons for not making this possible then I wonder what the closest possible pattern could be? I'm really keen to leverage all the amazing SQLAlchemy functionality which helps me access db records programatically from a class/object which I don't persist.

**Comment by zzzeek** [comment]
hi -

I dont understand what this would do (I can think of some versions of it, but you'd need to fill me in)


```py

class AbstractRelatedThing(Base):
    # abstract class.  never instantiated

class Thing(Base):
    __tablename__ = 'thing'

    # ...

    related = relationship(AbstractRelatedThing, ...)
```


so far that's what you're asking for ?

What happens when I do this?


```py


thing = session.get(Thing, 1)

# what does this do?
related_thing = thing.related


```


thanks for clarification

**Reply by gravy-jones-locker** [reply]
hi, thanks for the reply and sorry for the confusion - this is probably from my attempts to hack a solution.

I want to be able to do this

```python
class Thing(Base):
     __tablename__ = 'thing'
     
     id = Column(String, primary_key=True)
     
class AbstractThing(Base):
     __abstract__ = True
     
     thing_id = Column(ForeignKey('thing.id'))
     thing = relationship('Thing')  # With whatever necessary args to get it working
     
abstract_thing = AbstractThing(thing_id='foo')

# Access properties of thing with id == abstract_thing.thing_id
abstract_thing.thing.name
```

**Reply by gravy-jones-locker** [reply]
apologies if it seems insane. my best efforts at getting something like the pattern I want working

**Reply by zzzeek** [reply, marked as answer]
by definition [abstract classes are not instantiated](https://en.wikipedia.org/wiki/Abstract_type) so what you are requesting above is not supported.
